Sources/Property.swift

```diff
@@ -571,27 +571,28 @@ public final class Property<Value>: PropertyProtocol {
 					return observer.action(event)
 				}
 
-				box.lock.sync {
-					if let value = event.value {
-						box.value = value
+				box.modifyDidSet({ value in
+					if let newValue = event.value {
+						value = newValue
 					}
+				}, didSet: {
 					observer.action(event)
```

Comment:
It violates [access exclusivity](https://github.com/apple/swift-evolution/blob/master/proposals/0176-enforce-exclusive-access-to-memory.md) and the default copy-in, copy-out semantic of `inout`, which updates `_value` only upon the return of the closure. In-place mutation is an optimisation on top of these two assumptions.